python_modules/libraries/dagster-k8s/dagster_k8s/pipes.py

```diff
@@ -80,24 +83,86 @@ def read_messages(
         finally:
             self._handler = None
 
+    def _get_consume_logs_request_timeout(self) -> Optional[int]:
+        request_timeout_env_var = os.getenv("DAGSTER_K8S_PIPES_CONSUME_POD_LOGS_REQUEST_TIMEOUT")
+        if request_timeout_env_var:
+            return int(request_timeout_env_var)
+
+        # Timeout and reconnect every 10 minutes by default
+        return 600
```

Comment:
if we have to reprocess all the logs every time should we try to be more aggressive on how high we set this default? I worry it could get brutal for long running high log volume jobs

Reply:
1h maybe?

Reply:
this would mean that you'd have to wait a minimum of 1h before completing a pipes op?  That seems long enough that folks would regularly notice.

I'd vote for the timeout to be shorter and see if we could get the `since_time` (or `sinceTime`) param to work

Reply:
@prha the timeout doesn't change anything if the op completes successfully - it would be the maximum amount of time you would need to wait if this weird hanging connection issue happens